### PR TITLE
Backfill rake

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
Rework existing backfill script to make it more universal. 

## WHY
So that we can backfill any table that has missing `updated_at` values 

This is an action item from the Bigquery resiliency RFC. I'm tracking the status of sync migrations in a table here:

https://www.notion.so/quill/Airbyte-devops-cheat-sheet-07720c320c664681ae55c593d0b8dc7c

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  tested manually
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? |
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | (N/A or Yes)

### CODE REVIEW

File: services/QuillLMS/lib/tasks/backfill.rake [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    klass = args[:klass].constantize
+    fallback_fill_datetime = DateTime.new(2018,4,1)
+
+    fill_datetime = args[:fill_datetime] ? DateTime.parse(args[:fill_datetime]) : fallback_fill_datetime
+
+    records = klass.unscoped.where(updated_at: nil)
+
+    records.each do |record|
+      backfill_value = record.created_at || fill_datetime
+      puts "updating #{klass} with id #{record.id}"
```
Comment (dandrabik) [inline services/QuillLMS/lib/tasks/backfill.rake]: Should probably print `created_at` and `backfill_value` to track.

File: services/QuillLMS/lib/tasks/backfill.rake [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  desc 'backfills updated_at so null values do not exist'
+  task :updated_at, [:klass, :fill_datetime] => :environment do |t, args|
+    klass = args[:klass].constantize
+    fallback_fill_datetime = DateTime.new(2018,4,1)
+
+    fill_datetime = args[:fill_datetime] ? DateTime.parse(args[:fill_datetime]) : fallback_fill_datetime
+
+    records = klass.unscoped.where(updated_at: nil)
+
+    records.each do |record|
```
Comment (dandrabik) [inline services/QuillLMS/lib/tasks/backfill.rake]: Use `find_each` to more efficiently [fetch in batches](http://api.rubyonrails.org/classes/ActiveRecord/Batches.html#method-i-find_each).

File: services/QuillLMS/lib/tasks/backfill.rake [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+namespace :backfill do
+  # rake 'backfill:updated_at[ClassroomUnit "2020-01-01"]'
+  desc 'backfills updated_at so null values do not exist'
+  task :updated_at, [:klass, :fill_datetime] => :environment do |t, args|
+    klass = args[:klass].constantize
+    fallback_fill_datetime = DateTime.new(2018,4,1)
```
Comment (dandrabik) [inline services/QuillLMS/lib/tasks/backfill.rake:8]: This date seems arbitrary. Might be worth removing this fallback. Then if a fallback isn't passed in, it just doesn't update if there isn't a `created_at`.

**Review by dandrabik — APPROVED** [review]
So I left a few code comments. I see how this is needed for BigQuery, so approved for that use. Given this is generically written and mass overwrites data, we might consider removing this from the codebase after we migrate all of the necessary tables.

**Review by brendanshean — APPROVED** [review]

**Review by anathomical — APPROVED** [review]

**Review by anathomical — APPROVED** [review]

File: services/QuillLMS/lib/tasks/backfill.rake (hunk 3 above)
Comment (happythenewsad) [reply services/QuillLMS/lib/tasks/backfill.rake:8]: Fortunately, for the tables in question, all records have non-nil `created_at` values

**Pull request merged**